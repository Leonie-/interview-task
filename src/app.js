import React from "react";

import "./styles/reset.css";
import "./styles/index.css";

const App = ({ data }) => {
    return (
        <div className="interview">
            <h1>Fruit Selector</h1>
            <label htmlFor="fruit">Choose a fruit: </label>
            <select name="fruit" id="fruit" data-testid="fruit-select">
                <option value="one" data-testid="fruit-option">
                    One
                </option>
                <option value="two" data-testid="fruit-option">
                    Two
                </option>
                <option value="three" data-testid="fruit-option">
                    Three
                </option>
                <option value="four" data-testid="fruit-option">
                    Four
                </option>
            </select>
        </div>
    );
};

export { App };
