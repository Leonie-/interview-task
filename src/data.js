const data = [
    {
        type: "Pears",
        names: [
            {
                value: "Forelle",
                label: "Forelle",
                selected: false,
            },
            {
                value: "Anjou",
                label: "Anjou",
                selected: false,
            },
            {
                value: "Cornice",
                label: "Cornice",
                selected: false,
            },
            {
                value: "Clapp's Favourite",
                label: "Clapp's Favourite",
                selected: false,
            },
            {
                value: "Red Anjou",
                label: "Red Anjou",
                selected: false,
            },
        ],
        option: {
            value: "Pears",
            label: "Pears",
            selected: false,
        },
    },
    {
        type: "Apples",
        names: [
            {
                value: "Jonagold",
                label: "Jonagold",
                selected: false,
            },
            {
                value: "Golden Delicious",
                label: "Golden Delicious",
                selected: false,
            },
            {
                value: "Fuji",
                label: "Fuji",
                selected: false,
            },
            {
                value: "Pink Lady",
                label: "Pink Lady",
                selected: false,
            },
        ],
        option: {
            value: "Apples",
            label: "Apples",
            selected: false,
        },
    },
    {
        type: "Bananas",
        names: [
            {
                value: "Cavendish",
                label: "Cavendish",
                selected: false,
            },
        ],
        option: {
            value: "Bananas",
            label: "Bananas",
            selected: false,
        },
    },
    {
        type: "Strawberries",
        names: [
            {
                value: "Sweet Eve",
                label: "Sweet Eve",
                selected: false,
            },
            {
                value: "Veestar",
                label: "Veestar",
                selected: false,
            },
            {
                value: "Titan",
                label: "Titan",
                selected: false,
            },
            {
                value: "Musk",
                label: "Musk",
                selected: false,
            },
            {
                value: "Anitabis",
                label: "Anitabis",
                selected: false,
            },
        ],
        option: {
            value: "Strawberries",
            label: "Strawberries",
            selected: false,
        },
    },
    {
        type: "Pineapples",
        names: [
            {
                value: "Brecheche",
                label: "Brecheche",
                selected: false,
            },
            {
                value: "Smooth Cayenne",
                label: "Smooth Cayenne",
                selected: false,
            },
            {
                value: "Abacaxi",
                label: "Abacaxi",
                selected: false,
            },
        ],
        option: {
            value: "Pineapples",
            label: "Pineapples",
            selected: false,
        },
    },
];

export { data };
