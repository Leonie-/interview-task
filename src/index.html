<!doctype html>
<html lang="en" dir="ltr">
    <head>
        <meta charset="UTF-8">
        <title>Interview Task</title>
    </head>
    <body>
        <div id="app"></div>
    </body>
</html>