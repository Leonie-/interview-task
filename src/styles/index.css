body {
    background-color: #eee;
}
.interview {
    margin: 30px;
    padding: 20px;
    background-color: #fff;
}
