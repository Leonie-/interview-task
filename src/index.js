import React from "react";
import ReactDOM from "react-dom";
import { data } from "./data";
import { App } from "./app";
import "./styles/index.css";

// eslint-disable-next-line no-undef
const el = document.getElementById("app");
ReactDOM.render(<App data={data} />, el);
