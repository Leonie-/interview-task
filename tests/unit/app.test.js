import React from "react";
import { render, waitFor, fireEvent } from "@testing-library/react";

import { App } from "../../src/app";

describe("Metrics", () => {
    test("shows single dropdown with options", async () => {
        const { queryAllByTestId } = render(<App />);
        expect(queryAllByTestId("fruit-select").length).toBe(1);
        expect(queryAllByTestId("fruit-option").length).toBe(4);
    });

    test("renders correctly", async () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});
