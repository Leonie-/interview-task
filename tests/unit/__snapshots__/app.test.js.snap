// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics renders correctly 1`] = `
<DocumentFragment>
  <div
    class="interview"
  >
    <h1>
      Fruit Selector
    </h1>
    <label
      for="fruit"
    >
      Choose a fruit: 
    </label>
    <select
      data-testid="fruit-select"
      id="fruit"
      name="fruit"
    >
      <option
        data-testid="fruit-option"
        value="one"
      >
        One
      </option>
      <option
        data-testid="fruit-option"
        value="two"
      >
        Two
      </option>
      <option
        data-testid="fruit-option"
        value="three"
      >
        Three
      </option>
      <option
        data-testid="fruit-option"
        value="four"
      >
        Four
      </option>
    </select>
  </div>
</DocumentFragment>
`;
